File: JSONPlaces/Place.m
//
//  Place.m
//  JSONPlaces
//
//  Created by Adrian Kubała on 10.02.2017.
//  Copyright © 2017 Adrian. All rights reserved.
//

#import "Place.h"

@implementation Place

- (instancetype)initWithName:(NSString *)name latitude:(double)latitude longitude:(double)longitude pinImage:(UIImage *)pinImage {
  self = [super init];
  
  if (self) {
    _name = name;
    _location = [[CLLocation alloc] initWithLatitude:latitude longitude:longitude];
    _pinImage = pinImage;
  }
  
  return self;
}

- (instancetype)initWithName:(NSString *)name latitude:(double)latitude longitude:(double)longitude pinImage:(UIImage *)pinImage distance:(double)distance {
  self = [self initWithName:name latitude:latitude longitude:longitude pinImage:pinImage];
  
  if (self) {
    _distance = distance;
  }
  
  return self;
}

@end

File: Zadanie/Zadanie/Place.swift
//
//  PlacesStruct.swift
//  Zadanie
//
//  Created by Adrian on 12.08.2016.
//  Copyright © 2016 Adrian. All rights reserved.
//

import UIKit
import CoreLocation

struct Place {
  var name: String
  var coordinates: CLLocation
  var pinImage: UIImage?
  var distance: Double?
  
  init(name: String, latitude: Double, longitude: Double, pinImage: UIImage? = nil, distance: Double? = nil) {
    self.name = name
    coordinates = CLLocation(latitude: latitude, longitude: longitude)
    self.pinImage = pinImage
    self.distance = distance
  }
}
